Created onboarding screens that only show up when app is opened for the very first time

D-Ajay-Kumar opened the pull request:
…he first time

## Related Issue 

- The task was to create onboarding screens to give instructions to the user as to how to use the app's features.

Closes: #47 

### Describe the changes you've made

1. Created a set of onboarding screens with not much emphasis on the UI currently.
2. This set of onboarding screens appear only when the app is opened for the first time.
3. To check whether the app is opened the first time or not HiveDB is used to store the check.

## Type of change

What sort of change have you made:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Mention any unusual behaviour of your code (Write NA if not)
NA

## Checklist:
<!--
Example how to mark a checkbox:-
- [x] My code follows the code style of this project.
-->
- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly whereever it was hard to understand.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.

hiashutoshsingh commented on lib/pages/onboarding_screen.dart:
```diff
@@ -0,0 +1,51 @@
+
+class OnboardingScreen extends StatelessWidget {
+  final List<PageViewModel> listPageViewModels = [
+    PageViewModel(
```
Let me provide you with screenshots that will go here.

D-Ajay-Kumar replied:
sure

hiashutoshsingh commented on lib/pages/onboarding_screen.dart:
```diff
@@ -0,0 +1,51 @@
+        ),
+        onDone: () {
+          Navigator.pushReplacement(
+            context,
```
Navigation should be done using our routes mechanism. Please look at another place how we have done the project.

D-Ajay-Kumar replied:
Yeah I saw you guys are using GetIt for navigation

hiashutoshsingh commented on lib/ui/splash.dart:
```diff
@@ -14,12 +16,37 @@ class Splash extends StatefulWidget {
+  Box<bool> firstTimeBox;
+  bool firstTime;
+
+  Future<void> getHiveDate() async {
```
Please add comments here on what exactly this piece of code is doing.

hiashutoshsingh commented on lib/ui/splash.dart:
```diff
@@ -14,12 +16,37 @@ class Splash extends StatefulWidget {
+      () {
+        Navigator.pushReplacement(
+          context,
+          MaterialPageRoute(
```
Use Routes Navigation mechanism

hiashutoshsingh reviewed (changes requested):
I have added some comments which need a little change. Not some major changes though.

Let me know if something is not explained properly.

hiashutoshsingh commented:
@D-Ajay-Kumar Please also resolve this conflict.

D-Ajay-Kumar commented:
> @D-Ajay-Kumar Please also resolve this conflict.

Resolved the conflict, also added comments and changed routing mechanism. Could you provide with the screenshots of the onboarding screens or maybe create another issue for redesigning the screens according to provided pictures for other beginners?

hiashutoshsingh commented:
> > @D-Ajay-Kumar Please also resolve this conflict.
> 
> Resolved the conflict, also added comments and changed routing mechanism. Could you provide with the screenshots of the onboarding screens or maybe create another issue for redesigning the screens according to provided pictures for other beginners?

Sure.
@shagun25 Please provide him screenshots and he is blocked on this.

hiashutoshsingh commented:
@D-Ajay-Kumar hey, please take screenshots of the app from your side and place them in assets and use that.

D-Ajay-Kumar commented:
@hiashutoshsingh I am afraid I don't follow. I need to take screenshots of what screens of the app and use them as onboarding screens?

hiashutoshsingh commented:
> @hiashutoshsingh I am afraid I don't follow. I need to take screenshots of what screens of the app and use them as onboarding screens?

Sorry I wasn't clear enough.
Take screenshots of homescreen, sos, videos section, camera section..etc (if more can be added) from our app only.
so that user can know what kind of features are there in app.
@D-Ajay-Kumar

hiashutoshsingh commented:
@D-Ajay-Kumar Hey, any update on the above?

D-Ajay-Kumar commented:
Sorry about the delay in work. My college has announced surprise tests all of a sudden so I'm having direct most of my time there. Would be busy with that for 2 more days.

hiashutoshsingh commented:
> Sorry about the delay in work. My college has announced surprise tests all of a sudden so I'm having direct most of my time there. Would be busy with that for 2 more days.

ohhh OK. not an issue.

hiashutoshsingh commented:
@D-Ajay-Kumar any updates on the above, this is pending for a long.

Can you please resolve conflicts and attach screenshots.

Thanks
